FFI-exposed destroy_sector_builder function occasionally panics

We have seen cases in go-filecoin where a test which timed out causes a panic from rust-proofs. My suspicion is that this occurs when the SectorBuilder destructor is run after the process has already begun to shut down.

```
--- FAIL: TestNonceRace (30.99s)
    node_test.go:664: timed out waiting for commitSector message (for sector of size=127, from miner owner=fcqd85rxj2vjywk4l30uy7jzhjvtnmswcj43ev8c6) to appear in **miner** node's chain
FAIL
FAIL	github.com/filecoin-project/go-filecoin/node	47.781s
ok  	github.com/filecoin-project/go-filecoin/proofs	80.664s
ok  	github.com/filecoin-project/go-filecoin/protocol/hello	0.502s
ok  	github.com/filecoin-project/go-filecoin/protocol/retrieval	52.618s
ok  	github.com/filecoin-project/go-filecoin/protocol/storage	38.743s
ok  	github.com/filecoin-project/go-filecoin/repo	1.795s
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: "SendError(..)"', libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
thread '<unnamed>' panicked at '[SectorBuilder#drop] could not join sealer thread: Any', libcore/result.rs:1009:5
fatal runtime error: failed to initiate panic, error 5
SIGABRT: abort
PC=0x7fff6325eb66 m=10 sigcode=0

goroutine 0 [idle]:
runtime: unknown pc 0x7fff6325eb66
stack: frame={sp:0x700000b31948, fp:0x0} stack=[0x700000ab2290,0x700000b31e90)
0000700000b31848:  0000000000000000  0000000000000000 
0000700000b31858:  0000000000000000  0000000000000000 
0000700000b31868:  0000000000000000  0000000000000000 
0000700000b31878:  0000000000000000  0000700000b31940 
0000700000b31888:  0000000004e59f38  0000000000000050 
0000700000b31898:  0000000000000000  0000000000000003 
0000700000b318a8:  0000000000000000  0000000000000003 
0000700000b318b8:  0000700000b31980  0000000004ebcd78 
0000700000b318c8:  0000700000b31a00  0000700000b31a10 
0000700000b318d8:  0000700000b31a00  0000000000000001 
0000700000b318e8:  0000002000000000  0000000000001e03 
0000700000b318f8:  0000000004ebce68  0000000004ebce68 
0000700000b31908:  0000000004e7eea8  0000000004ebce68 
0000700000b31918:  0000000000000001  0000000000000000 
0000700000b31928:  0000000004e7f0c8  0000000004ebcfa0 
0000700000b31938:  0000700000b31b28  0000700000b319c0 
0000700000b31948: <00007fff63429080  0000700000b32000 
0000700000b31958:  0000700000b31998  0000000000000000 
0000700000b31968:  0000000004e7f0c8  0000000004ebcfa0 
0000700000b31978:  0000700000b31b28  0000700000b319b0 
0000700000b31988:  00007fff631ba1ae  0000000004e7f057 
0000700000b31998:  00000000ffffffdf  ffffffff00b319c0 
0000700000b319a8:  0000000000000001  0000700000b319c0 
0000700000b319b8:  0000000004e41759  0000700000b31a10 
0000700000b319c8:  0000000004e46abb  0000000004ebce48 
0000700000b319d8:  0000000000000002  0000000004e7ee68 
0000700000b319e8:  0000000000000001  0000700000b31a00 
0000700000b319f8:  0000000000000001  0000700000b31a28 
0000700000b31a08:  0000000004e5bff0  0000700000b31a80 
0000700000b31a18:  0000000004e489de  000000000d049640 
0000700000b31a28:  0000000004ebd098  0000000000000001 
0000700000b31a38:  0000000004e7f290  0000000000000001 
runtime: unknown pc 0x7fff6325eb66
stack: frame={sp:0x700000b31948, fp:0x0} stack=[0x700000ab2290,0x700000b31e90)
0000700000b31848:  0000000000000000  0000000000000000 
0000700000b31858:  0000000000000000  0000000000000000 
0000700000b31868:  0000000000000000  0000000000000000 
0000700000b31878:  0000000000000000  0000700000b31940 
0000700000b31888:  0000000004e59f38  0000000000000050 
0000700000b31898:  0000000000000000  0000000000000003 
0000700000b318a8:  0000000000000000  0000000000000003 
0000700000b318b8:  0000700000b31980  0000000004ebcd78 
0000700000b318c8:  0000700000b31a00  0000700000b31a10 
0000700000b318d8:  0000700000b31a00  0000000000000001 
0000700000b318e8:  0000002000000000  0000000000001e03 
0000700000b318f8:  0000000004ebce68  0000000004ebce68 
0000700000b31908:  0000000004e7eea8  0000000004ebce68 
0000700000b31918:  0000000000000001  0000000000000000 
0000700000b31928:  0000000004e7f0c8  0000000004ebcfa0 
0000700000b31938:  0000700000b31b28  0000700000b319c0 
0000700000b31948: <00007fff63429080  0000700000b32000 
0000700000b31958:  0000700000b31998  0000000000000000 
0000700000b31968:  0000000004e7f0c8  0000000004ebcfa0 
0000700000b31978:  0000700000b31b28  0000700000b319b0 
0000700000b31988:  00007fff631ba1ae  0000000004e7f057 
0000700000b31998:  00000000ffffffdf  ffffffff00b319c0 
0000700000b319a8:  0000000000000001  0000700000b319c0 
0000700000b319b8:  0000000004e41759  0000700000b31a10 
0000700000b319c8:  0000000004e46abb  0000000004ebce48 
0000700000b319d8:  0000000000000002  0000000004e7ee68 
0000700000b319e8:  0000000000000001  0000700000b31a00 
0000700000b319f8:  0000000000000001  0000700000b31a28 
0000700000b31a08:  0000000004e5bff0  0000700000b31a80 
0000700000b31a18:  0000000004e489de  000000000d049640 
0000700000b31a28:  0000000004ebd098  0000000000000001 
0000700000b31a38:  0000000004e7f290  0000000000000001 

goroutine 18 [syscall]:
runtime.cgocall(0x44c7e00, 0xc0000446c0, 0x337b)
	/usr/local/go/src/runtime/cgocall.go:128 +0x5e fp=0xc000044690 sp=0xc000044658 pc=0x4004efe
github.com/filecoin-project/go-filecoin/sectorbuilder._Cfunc_destroy_sector_builder(0xd046ec0)
	_cgo_gotypes.go:387 +0x41 fp=0xc0000446c0 sp=0xc000044690 pc=0x44ba791
github.com/filecoin-project/go-filecoin/sectorbuilder.(*RustSectorBuilder).destroy.func1(0xd046ec0)
	/Users/distiller/go/src/github.com/filecoin-project/go-filecoin/sectorbuilder/rustsectorbuilder.go:293 +0x56 fp=0xc0000446f8 sp=0xc0000446c0 pc=0x44c6276
github.com/filecoin-project/go-filecoin/sectorbuilder.(*RustSectorBuilder).destroy(0xc00001fc20)
	/Users/distiller/go/src/github.com/filecoin-project/go-filecoin/sectorbuilder/rustsectorbuilder.go:293 +0x2f fp=0xc000044710 sp=0xc0000446f8 pc=0x44bc65f
github.com/filecoin-project/go-filecoin/sectorbuilder.NewRustSectorBuilder.func7(0xc00001fc20)
	/Users/distiller/go/src/github.com/filecoin-project/go-filecoin/sectorbuilder/rustsectorbuilder.go:115 +0x2b fp=0xc000044728 sp=0xc000044710 pc=0x44c555b
runtime.call32(0x0, 0x4727a70, 0xc00046a010, 0x1000000010)
	/usr/local/go/src/runtime/asm_amd64.s:522 +0x3b fp=0xc000044758 sp=0xc000044728 pc=0x405be9b
runtime.runfinq()
	/usr/local/go/src/runtime/mfinal.go:222 +0x1d2 fp=0xc0000447e0 sp=0xc000044758 pc=0x40196c2
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc0000447e8 sp=0xc0000447e0 pc=0x405db91
created by runtime.createfing
	/usr/local/go/src/runtime/mfinal.go:156 +0x61
```

--- a/filecoin-proofs/src/api/sector_builder/mod.rs
+++ b/filecoin-proofs/src/api/sector_builder/mod.rs
@@ -22,10 +22,6 @@ const NUM_SEAL_WORKERS: usize = 2;
 
 const FATAL_NOSEND_TASK: &str = "[run_blocking] could not send";
 const FATAL_NORECV_TASK: &str = "[run_blocking] could not recv";
-const FATAL_KILL_SCHDLR: &str = "[SectorBuilder#drop] could not send Shutdown";
-const FATAL_KILL_SEALER: &str = "[SectorBuilder#drop] could not send Shutdown to sealer";
-const FATAL_JOIN_SCHDLR: &str = "[SectorBuilder#drop] could not join scheduler thread";
-const FATAL_JOIN_SEALER: &str = "[SectorBuilder#drop] could not join sealer thread";
 
 pub type SectorId = u64;
 
@@ -157,26 +153,32 @@ impl SectorBuilder {
 impl Drop for SectorBuilder {
     fn drop(&mut self) {
         // Shut down main worker and sealers, too.
-        self.scheduler_tx
+        let _ = self
+            .scheduler_tx
             .send(Request::Shutdown)
-            .expect(FATAL_KILL_SCHDLR);
+            .map_err(|err| println!("err sending Shutdown to scheduler: {:?}", err));
 
         for _ in &mut self.sealers {
-            self.sealers_tx
+            let _ = self
+                .sealers_tx
                 .send(SealerInput::Shutdown)
-                .expect(FATAL_KILL_SEALER);
+                .map_err(|err| println!("err sending Shutdown to sealer: {:?}", err));
         }
 
         // Wait for worker threads to return.
         let scheduler_thread = &mut self.scheduler.thread;
 
         if let Some(thread) = scheduler_thread.take() {
-            thread.join().expect(FATAL_JOIN_SCHDLR)
+            let _ = thread
+                .join()
+                .map_err(|err| println!("err joining scheduler thread: {:?}", err));
         }
 
         for worker in &mut self.sealers {
             if let Some(thread) = worker.thread.take() {
-                thread.join().expect(FATAL_JOIN_SEALER)
+                let _ = thread
+                    .join()
+                    .map_err(|err| println!("err joining sealer thread: {:?}", err));
             }
         }
     }

